### PR TITLE
S28-3072/ams generate asset

### PR DESCRIPTION
**Pull request by jasonpaige**
### JIRA ticket(s)

- S28-3072

### Change description

- This PR adds an endpoint to be called by the current video editing process so that video assets can be generated for an existing mp4 video which has been uploading to a storage account blob container manually.

> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**

- Create a new container in the prefinalsastg (any name is fine)
- Add a sample mp4 video to this new container
- Make another container in prefinalsastg
- Create a test user using https://pre-api-pr-585.dev.platform.hmcts.net/swagger-ui/index.html
- Call the POST [/media-service/generate-asset](https://pre-api-pr-585.dev.platform.hmcts.net/swagger-ui/index.html#/media-service-controller/generateAsset) endpoint with the app id returned when you created the test user on the previous step and for the code use `SecureKey`. For the body:
```json
{
  "description": "<some test description>",
  "final_asset": "<any unique name>",
  "source_container": "<name of the first container you created>",
  "destination_container": "<name of last container you created>",
  "temp_asset": "<any unique name>"
}
```
- you should expect a 200 result and the destination container should be populated with asset files.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java [hunk 1]
```diff
@@ -199,4 +214,25 @@ public ResponseEntity<CaptureSessionDTO> startLiveEvent(@PathVariable UUID captu
...
+    @PostMapping("/generate-asset")
+    @Operation(
+        operationId = "generateAsset",
+        summary = "LEGACY - Given a source & destination, this endpoint will generate a streaming asset for a given mp4"
+    )
+    @Parameter(
+        name = "code",
+        description = "Rudimentary security code to prevent unauthorised access to this endpoint",
+        schema = @Schema(implementation = String.class)
+    )
```
Comment (lucas-phillips28) [inline src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java:227]: Does this mean it needs to be ignored by the X-User-Id auth checker?

**Review by lucas-phillips28 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java (hunk 1 above)
Comment (jasonpaige) [reply src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java:227]: yes it does as it'll get called as part of the manual editing process which won't have an X-User-Id
Is there something I need to change in addition to get that to work

Comment (lucas-phillips28) [reply src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java:227]: Yeah- you just need to add the endpoint to this list: https://github.com/hmcts/pre-api/blob/master/src/main/java/uk/gov/hmcts/reform/preapi/config/SecurityConfig.java#L24-L38

File: src/main/java/uk/gov/hmcts/reform/preapi/config/SecurityConfig.java [hunk 2]
```diff
@@ -36,6 +36,7 @@ public class SecurityConfig {
         new AntPathRequestMatcher("/error"),
         new AntPathRequestMatcher("/invites", "GET"),
         new AntPathRequestMatcher("/invites/redeem", "POST"),
+        new AntPathRequestMatcher("/media/generate-asset", "POST"),
```
Comment (lucas-phillips28) [inline src/main/java/uk/gov/hmcts/reform/preapi/config/SecurityConfig.java:39]:
```suggestion
        new AntPathRequestMatcher("/media-service/generate-asset", "POST"),
```

**Review by lucas-phillips28 — APPROVED** [review]

**Pull request merged**